src/Header.cpp

```diff
@@ -0,0 +1,37 @@
+#include "lift/Header.hpp"
+
+#include <string>
+
+namespace lift {
+
+Header::Header(
+    std::string_view name,
+    std::string_view value)
+{
+    m_header.reserve(name.length() + value.length() + 2);
+    m_header.append(name);
+    m_header.append(": ");
+    m_header.append(value);
+
+    auto colon_pos = m_header.find(':');
+    if (colon_pos != std::string_view::npos) {
+        /**
+         * Use remove_suffix|prefix for safety bounds checking.
+         */
+
+        m_name = m_header;
+        m_value = m_header;
+
+        // Remove the value from the name.
+        m_name.remove_suffix(m_header.length() - colon_pos);
+
+        // Remove the name from the value.
+        m_value.remove_prefix(colon_pos + 1);
+        // Remove any prefix whitespace.
+        while (!m_value.empty() && std::isspace(m_value[0]) != 0) {
```

Comment:
if we're removing arbitrary whitespace from the value, shouldn't we do the same for the name?

Reply:
I think maybe we should just get rid of the whitespace strip actually and do `remove_prefix(m_colon_position + 2)`